x-pack/plugins/fleet/server/services/epm/package_service.mock.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { PackageClient, PackageService } from './package_service';
+
+const createClientMock = (): jest.Mocked<PackageClient> => ({
+  getInstallation: jest.fn(),
+  ensureInstalledPackage: jest.fn(),
+  fetchFindLatestPackage: jest.fn(),
+  getRegistryPackage: jest.fn(),
+  reinstallEsAssets: jest.fn(),
+});
+
+const createServiceMock = (): jest.Mocked<PackageService> => ({
+  asScoped: jest.fn().mockReturnValue(createClientMock()),
```

Comment:
Suggestion: instead of returning a shared Package client, use a function in the `fn()` to return a new object every time:

```suggestion
  asScoped: jest.fn(() => createClientMock()),
```